src/client/app/components/LoginComponent.tsx

```diff
@@ -2,39 +2,89 @@
  * License, v. 2.0. If a copy of the MPL was not distributed with this
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
-import React from 'react';
+import * as React from 'react';
 import axios from 'axios';
 import { browserHistory } from 'react-router';
+import { Notification } from 'react-notification-system';
 import HeaderComponent from '../components/HeaderComponent';
+import { ShowNotificationAction } from '../actions/notifications';
 import FooterComponent from '../components/FooterComponent';
 
 
-export default class LoginComponent extends React.Component {
+interface LoginProps {
+	showNotification(notification: Notification): ShowNotificationAction;
+}
+
+interface LoginState {
+	email: string;
+	password: string;
+}
+
+export default class LoginComponent extends React.Component<LoginProps, LoginState> {
+	private inputEmail: HTMLInputElement | null;
 	/**
 	 * Initializes the component's state to include email (email users use to login) and password (corresponding to their email)
 	 * Binds the functions to 'this' LoginComponent
 	 */
-	constructor(props) {
+	constructor(props: LoginProps) {
 		super(props);
 		this.state = { email: '', password: '' };
 		this.handleEmailChange = this.handleEmailChange.bind(this);
 		this.handlePasswordChange = this.handlePasswordChange.bind(this);
 		this.handleSubmit = this.handleSubmit.bind(this);
 	}
 
+	/**
+	 * @return JSX to create the login panel
+	 */
+	public render() {
+		const formStyle = {
```

Comment:
Shouldn't these have `React.CSSProperties`

Reply:
I generally took the view that when types can be inferred they should be, as is the case here. However, that's certainly a decision we can talk about changing.